croptalk/tools.py

```diff
@@ -52,29 +69,44 @@ def get_sp_document(state: Optional[str] = None,
 
         # Define your SQL query
         sql_query = f"""
-        SELECT s3_key 
-        FROM sp_files
-        WHERE year = {year} AND 
-        commodity_name = '{commodity.lower()}' AND
-        state_name = '{state.lower()}' AND 
-        county_name = '{county.lower()}'
-        """
+           SELECT s3_key, year
+           FROM sp_files
+           WHERE commodity_name = '{commodity.lower()}' AND
+           state_name = '{state.lower()}' AND 
+           county_name = '{county.lower()}'
+           """
 
         # Execute the SQL query
         with engine.connect() as connection:
             result = connection.execute(text(sql_query))
             rows = result.fetchall()
 
-        if not rows:
-            return (f"My search results indicate that there is no SP document for the corresponding year ({year}), "
+        s3_keys = pd.DataFrame(rows)
+        if s3_keys.empty:
+            return (f"My search results indicate that there is no SP document for the corresponding "
                     f"commodity ({commodity}), state ({state}),and county ({county}). \b"
-                    "Make sure you are providing available year, commodity, state and county.")
-        return (f"Here is the link to the SP document you're looking for : "
-                f"https://croptalk-spoi.s3.us-east-2.amazonaws.com/SPOI/{rows[0][0]}")
+                    "Make sure you are providing available commodity, state and county.")
+
+        if not year:
+            # take document in latest year
+            s3_keys = s3_keys[s3_keys["year"] == s3_keys["year"].max()]
+        else:
+            s3_keys = s3_keys[s3_keys["year"] == year]
+
+        doc_link = s3_keys["s3_key"].values[0]
```

Comment:
I don't think we are covering the case where a specific year was specified, but only different years were found in the DB... do we?